### PR TITLE
scan: pagination system 

### LINKED ISSUE #750: Implement pagination component
Include state and guideline on how to use them on other pages

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #750 

- [x] Implement Pagination Component
- [x] Use on BlockHomePage (For example)

### CODE REVIEW

File: scan/src/utils/Page.re [hunk 1]
```diff
@@ -0,0 +1,6 @@
+let findNumberOfPage = (amount, limit) =>
```
Comment (sorawit) [inline scan/src/utils/Page.re]: So let's name this `getPageCount = (total, limit) => ...`

File: scan/src/components/Pagination.re [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    <img
+      src=Images.leftAngle
+      className={Css.merge([Styles.angle(isPrevious), Styles.clickable(active)])}
+      onClick
+    />;
+  };
+};
+
+[@react.component]
+let make = (~currentPage, ~pageCount, ~onChangePage: int => unit) => {
```
Comment (sorawit) [inline scan/src/components/Pagination.re]: `onPageChange`

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/components/Pagination.re [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      display(`flex),
+      flexDirection(`row),
+      width(`percent(100.)),
+      justifyContent(`center),
+      minHeight(`px(30)),
+    ]);
+
+  let innerContainer = style([display(`inlineFlex), alignItems(`center)]);
+
+  let angle = isflip =>
```
Comment (taobun) [inline scan/src/components/Pagination.re]: isFlip

File: scan/src/pages/BlockHomePage.re [hunk 4]
```diff
@@ -69,13 +69,27 @@ let renderBody = ((block, proposer): (BlockHook.Block.t, option(ValidatorHook.Va
...
+    let%Opt blocks = blocksCountOpt;
+    let%Opt latestBlock = blocks->Belt.List.get(0);
+    Some(latestBlock.height);
+  };
+  let pageCount =
+    {
+      let%Opt latestHeight = latestHeightOpt;
+      Some(Page.getPageCount(latestHeight, limit));
+    }
+    |> Belt.Option.getWithDefault(_, 1);
```
Comment (sorawit) [inline scan/src/pages/BlockHomePage.re:89]: We will need to revisit this sometime after we clearly define how to deal with loading state.

**Review by sorawit — DISMISSED** [review]

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/pages/BlockHomePage.re (hunk 4 above)
Comment (evilpeach) [reply scan/src/pages/BlockHomePage.re:89]: Sure

**Review by taobun — APPROVED** [review]

**Review by prin-r — APPROVED** [review]

**Pull request merged**